docs/design/design.md

```diff
@@ -4,81 +4,95 @@ Summary of our design decisions and some pointers to the literature.
 
 Let's talk about just Python: the wrappers are in the [symengine.py project](https://github.com/symengine/symengine.py). They are implemented using Cython, and they are free to introduce any kind of classes (including SymPy's `Expr` or Sage's `Expression` if needed), and the point of the wrappers is to make sure that things work out of the box from SymPy and Sage. The only job of the C++ SymEngine library is to ensure that the library's C++ API is implemented in such a way so that the wrappers can be written to do what they need. For example, we could easily introduce SymPy's `Expr` into the wrappers, by simply introducing the `Expr` class and make all the other classes subclass from it instead of from `Basic`.
 
-That was the reason we split the wrappers, so now in the (pure) C++ `symengine/symengine` repository, we only have to worry about speed, correctness, maintainability and a usable API, and we can concentrate on these things without worrying or even testing any kind of wrappers. In the wrappers (`symengine/symengine.py`, or `.jl`, `.rb`), we simply just use the C++ (or C) API and the only thing we care is so that the (Python) wrapper can be used from sympy/Sage (and we test that in the test suite), and that it doesn't introduce unnecessary overhead in terms of speed. Ruby or Julia wrappers then care about interoperability with other libraries in those languages.
+That was the reason we split the wrappers, so now in the (pure) C++ `symengine/symengine` repository, we only have to worry about speed, correctness, maintainability and a usable API, and we can concentrate on these things without worrying or even testing any kind of wrappers. In the wrappers (`symengine/symengine.py`, or `.jl`, `.rb`), we simply just use the C++ (or C) API and the only thing we care is so that the (Python) wrapper can be used from sympy/Sage (and we test that in the test suite), and that it doesn't introduce unnecessary overhead in terms of speed. Ruby or Julia (or R) wrappers then care about interoperability with other libraries in those languages.
 
-* [https://github.com/symengine/symengine.py](https://github.com/symengine/symengine.py)
-* <https://github.com/symengine/symengine.rb>
-* <https://github.com/symengine/SymEngine.jl>
+- [https://github.com/symengine/symengine.py](https://github.com/symengine/symengine.py)
+- <https://github.com/symengine/symengine.rb>
+- <https://github.com/symengine/SymEngine.jl>
+- <https://github.com/symengine/symengine.R>
 
 ## Reference Counted Pointers
 
 ### Teuchos `RCP`
 
 Memory management is handled by RCP (reference counted pointers) from
-`Trilinos` (module `Teuchos`). We have copied the relevant files into
-`src/teuchos`, so no external dependency is needed.  Brief code snippets of the
+`Trilinos` (the module `Teuchos`). We have copied the relevant files into
+`src/teuchos`, so no explicit dependency is present. Brief code snippets of the
```

Comment:
`explicit dependency is present`? I don't think this change is correct.

Reply:
Yup, this was a poor choice of words, sorry. Does `so no external dependencies need to be obtained.` sound better?